### DISCUSSION TITLE
How to get bucket location in AWS SDK 2.x

### DISCUSSION THREAD

**Discussion by Fais-007** [aws/aws-sdk-java-v2#3606] [Q&A]
I am trying to get bucket region. I only have bucket name, access key and secret key. How do I find out which region this bucket belongs to?

**Comment by debora-ito** [comment]
There's no S3 API that will return the bucket region as a response field.

What you can do is call HeadBucket and check the response for the `x-amz-bucket-region` header, but note that if the bucket exists and it's in the same region endpoint you sent the request to, the action returns a 200 OK, if the bucket is in a different region the response is a 301 Moved Permanently.

Here's an example in wirelogs (I removed some of the headers to make it easier to read):
```
>> "HEAD / HTTP/1.1[\r][\n]"
>> "Host: this-bucket-is-in-ap-mumbai.s3.us-west-2.amazonaws.com[\r][\n]"
>> "amz-sdk-invocation-id: f268e8c9-f617-3179-2fa9-add9f800dfda[\r][\n]"
>> "amz-sdk-request: attempt=1; max=4[\r][\n]"
>> "X-Amz-Date: 20221215T013720Z[\r][\n]"
>> "Connection: Keep-Alive[\r][\n]"
>> "[\r][\n]"
<< "HTTP/1.1 301 Moved Permanently[\r][\n]"
<< "x-amz-bucket-region: ap-south-1[\r][\n]"
<< "x-amz-request-id: GQQ2Y9S3AP53AZ39[\r][\n]"
<< "Content-Type: application/xml[\r][\n]"
<< "Date: Thu, 15 Dec 2022 01:37:20 GMT[\r][\n]"
<< "Server: AmazonS3[\r][\n]"
<< "[\r][\n]"
```

The s3 client was set to use `us-west-2` and the bucket is in `ap-south-1`. The `"x-amz-bucket-region` response header is `ap-south-1`.
